### PR TITLE
Add missing TEST_OUTPUT sections to most tests in fail_compilation

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
These test were simply missing an appropriate section. The remaining ones need
some extensions to d_do_test.d to handle e.g. paths.

### CODE REVIEW

File: src/dmd/hdrgen.d [hunk 1]
```diff
@@ -3107,10 +3107,14 @@ private void sizeToBuffer(Expression e, OutBuffer* buf, HdrGenState* hgs)
 {
     if (e.type == Type.tsize_t)
     {
-        Expression ex = (e.op == TOK.cast_ ? (cast(CastExp)e).e1 : e);
+        auto ce = e.isCastExp();
+        Expression ex = ce ? ce.e1 : e;
         ex = ex.optimize(WANTvalue);
-        const dinteger_t uval = ex.op == TOK.int64 ? ex.toInteger() : cast(dinteger_t)-1;
-        if (cast(sinteger_t)uval >= 0)
+
```
Comment (RazvanN7) [inline src/dmd/hdrgen.d]: This change should be the subject a different PR. There is no mention about this in the title or description.

File: test/fail_compilation/fail19890a.d [hunk 2]
```diff
@@ -1,7 +1,9 @@
-// PERMUTE_ARGS:
 /*
+PERMUTE_ARGS:
+TEST_OUTPUT:
 ---
-fail_compilation/fail19890a.d(8): Error: `void[/^[0-9]+(LU)?$/]` size 1 * /^[0-9]+$/ exceeds 0x7fffffff size limit for static array
+fail_compilation/fail19890a.d(9): Error: `void[$n$]` size 1 * $n$ exceeds 0x7fffffff size limit for static array
```
Comment (RazvanN7) [inline test/fail_compilation/fail19890a.d]: There is no mention of this change in the description or title.

File: test/fail_compilation/fail19890b.d [hunk 3]
```diff
@@ -1,7 +1,9 @@
-// PERMUTE_ARGS:
 /*
+PERMUTE_ARGS:
+TEST_OUTPUT:
 ---
-fail_compilation/fail19890b.d(8): Error: `void[/^[0-9]+(LU)?$/]` size 1 * /^[0-9]+$/ exceeds 0x7fffffff size limit for static array
+fail_compilation/fail19890b.d(9): Error: `void[$n$]` size 1 * $n$ exceeds 0x7fffffff size limit for static array
```
Comment (RazvanN7) [inline test/fail_compilation/fail19890b.d]: ditto

File: test/fail_compilation/staticarrayoverflow.d [hunk 4]
```diff
@@ -7,7 +7,7 @@ fail_compilation/staticarrayoverflow.d(24): Error: static array `S[1879048192]`
 fail_compilation/staticarrayoverflow.d(24): Error: variable `staticarrayoverflow.y` size overflow
 fail_compilation/staticarrayoverflow.d(26): Error: static array `S[8070450532247928832]` size overflowed to 8070450532247928832
 fail_compilation/staticarrayoverflow.d(26): Error: variable `staticarrayoverflow.a` size overflow
-fail_compilation/staticarrayoverflow.d(27): Error: static array `S[0][18446744073709551615LU]` size overflowed to 18446744073709551615
+fail_compilation/staticarrayoverflow.d(27): Error: static array `S[0][18446744073709551615]` size overflowed to 18446744073709551615
```
Comment (RazvanN7) [inline test/fail_compilation/staticarrayoverflow.d]: ditto

**Review by RazvanN7 — CHANGES_REQUESTED** [review]
Please document all changes and add a proper title and description for the PR.

**Comment by Geod24** [comment]
@MoonlightSentinel : Now that https://github.com/dlang/dmd/pull/10607 is in, I think it makes sense to submit a patch that removes `LU` from the literals (everywhere).

One reason to have `LU` was (IMO) to allow for code to be copy-pasted, because having error message spew up code that doesn't compile isn't great. But after the aforementioned PR, it would make sense to remove `L`/`LU` (and also help a lot with error messages).

**Comment by MoonlightSentinel** [comment]
> Please document all changes and add a proper title and description for the PR.

Sorry if the commit title was not sufficient but these changes were just an experiment to remove this inconsistency and not intended to remain. I just did not remove them yet

File: test/fail_compilation/fail19890a.d (hunk 2 above)
Comment (MoonlightSentinel) [reply test/fail_compilation/fail19890a.d]: This comment was just an informal description of the error message and not checked at all. The $n$ is necessary because the number depends on the test platform

**Comment by MoonlightSentinel** [comment]
The second commit is just a temporary aid to find `compilable` tests without an appropriate section.

File: test/fail_compilation/fail_arrayop3a.d [hunk 5]
```diff
@@ -1,23 +1,14 @@
-// REQUIRED_ARGS: -o-
-
 /*
-example output from druntime
+REQUIRED_ARGS: -o-
+TEST_OUTPUT:
 ----
-../../druntime/import/core/internal/arrayop.d(160): Error: static assert  "Binary op `*` not supported for element type X."
```
Comment (MoonlightSentinel) [inline test/fail_compilation/fail_arrayop3a.d]: Note that this _output_ is just an informal comment and was never enforced by the test suite.
Ditto for the other `fail_arrayop` tests.

**Review by wilzbach — APPROVED** [review]

**Comment by MoonlightSentinel** [comment]
This should be good once #10831 is merged.

**Review by RazvanN7 — APPROVED** [review]
